Feature/kas 3933 link user and organisations with mandataris

StuyckRuben opened the pull request:
https://kanselarij.atlassian.net/browse/KAS-3933
Add organization detail page
Add functionality to link organization to mandatee

- [ ] Merge https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/386

Commit:
Link users to mandatees

Commit:
add custom mandatee-person-selector component
add mandatee-person-selector to user detail page
add functionality to link user to mandatee
add organization detail page
add functionality to link organization to mandatees

sergiofenoll commented on app/components/mandatees/mandatee-person-selector.js:
This component is a bit strange, because it doesn't seem to take the `dateRange` into account *except* when searching. So if the dateRange is set to e.g. 2016 the list still shows the ministers from the current government.

But this behaviour is the same as when it was implemented in the `GenerateReportModal`, so it can stay as is here and we can have a look at it later.

sergiofenoll commented on app/routes/settings/users/user.js:
In the user detail screen, we show the linked mandatee in the dropdown. We use `selectedPerson` to display the link/unlink button. You need to set the value of `selectedPerson` with `model.person` so that works on a new page load, otherwise we will show the selector with the linked mandatee but we won't show an unlink button.

You can do this by expanding the route class like this:

```diff
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';

 export default class SettingsUsersUserRoute extends Route {
   @service store;

+  person;

   model(params) {
     return this.store.findRecord('user', params.id, {
       include: 'memberships.organization,memberships.role,person'
     });
   }
+
+  async afterModel(model) {
+    this.person = await model.person;
+  }
+
+  setupController(controller, model) {
+    super.setupController(controller, model);
+    controller.selectedPerson = this.person;
+  }
 }
```

This will load in the `person` relationship after the model has been loaded, and it will then assign it to a `this.person` member of the route. Afterwards, during the `setupController` hook, we assign the person to the controller's `selectedPerson` member. This way it's pre-loaded when we display the route.

sergiofenoll commented on app/templates/settings/organizations/index.hbs:
```diff
@@ -123,7 +123,22 @@
                   {{t "block-organization"}}
                 </AuButton>
               {{/if}}
+              <AuButton
+              @skin="naked"
+              role="menuitem"
+              {{on "click" (queue (toggle "showMandateeSelectorModal" this) (set this "organizationBeingLinked" organization))}}
+              >
+                {{t "connect to mandatee"}}
+              </AuButton>
```
This seems unused.

sergiofenoll commented on app/templates/settings/users/user.hbs:
```diff
@@ -130,6 +130,45 @@
+              </Group>
+              </AuToolbar>
+          </Auk::Panel::Header>
+          <Auk::Panel::Body class="au-u-padding-none">
```
It looks a little better if you leave the default padding in.

ValenberghsSven replied:
hmm, somewhat disagree. No padding means the borders match up. 
![image](https://user-images.githubusercontent.com/22245223/233616909-988e8ce9-ce5e-489f-aaf4-b8caa63fa402.png)


Default padding means floating lines

![image](https://user-images.githubusercontent.com/22245223/233616842-abbee18f-95fb-44ad-8edb-a1c73ab9d293.png)


But styling without a prototype/design is just exchanging opinions.
This can be fine-tuned later.

sergiofenoll replied:
That's a different view. I'm talking about the view in the user detail. You're showing the organization detail.

No padding:
![image](https://user-images.githubusercontent.com/22411874/233618731-74618446-c9b9-4808-85cc-a0e37a4f55af.png)

With padding:
![image](https://user-images.githubusercontent.com/22411874/233618864-14f55dc3-238d-4bf2-a521-54cf331b06b0.png)

Commit:
remove unused button on organizatinos index template
add default padding in the user template on the mandatee selector
add pre-loading of the selectedPerson in the user route

ValenberghsSven committed:
KAS-3933 some cleanup and prettier

ValenberghsSven committed:
KAS-3933 add serializer to prevent overwriting login-activity and account

ValenberghsSven committed:
KAS-3933 correct label for minister, make selector label hideable

ValenberghsSven reviewed (approved):
Made some changes, mostly prettier and some styling / cleanup
Added a serializer for user to avoid saving the `login-activity` (done in microservice) and `account` (should not be able to be updated or saved in frontend)

ValenberghsSven committed:
KAS-3933 remove unused translation

ValenberghsSven commented:
Merged this before jenkins was done.
42 of 58 already run with 0 failed tests.
The other tests are mainly publication/search.
Want to cut this process short to get a dev yggie rebuild/reindex going with other merged PR before the weekend.
